tools/cli/commands/release.js

```diff
@@ -0,0 +1,200 @@
+/**
+ * External dependencies
+ */
+import chalk from 'chalk';
+import child_process from 'child_process';
+import inquirer from 'inquirer';
+
+/**
+ * Internal dependencies
+ */
+import promptForProject from '../helpers/promptForProject';
+import { chalkJetpackGreen } from '../helpers/styling';
+import { allProjects } from '../helpers/projectHelpers';
+
+/**
+ * Command definition for the release subcommand.
+ *
+ * @param {object} yargs - The Yargs dependency.
+ * @returns {object} Yargs with the build commands defined.
+ */
+export function releaseDefine( yargs ) {
+	yargs.command(
+		'release [project] [script]',
+		'Runs a release script for a project.',
+		yarg => {
+			yarg
+				.positional( 'project', {
+					describe:
+						'Project in the form of type/name, e.g. plugins/jetpack, or type, e.g. plugins.',
+					type: 'string',
+				} )
+				.positional( 'script', {
+					alias: 's',
+					describe: 'The release script to run',
+					type: 'string',
+					choices: [ 'changelog', 'readme', 'release-branch', 'amend' ],
+				} )
+				.option( 'beta', {
+					alias: 'b',
+					describe: 'Is this a beta?',
+					type: 'boolean',
+					default: false,
+				} );
+		},
+		async argv => {
+			await releaseCli( argv );
+			if ( argv.v ) {
+				console.log( argv );
+			}
+		}
+	);
+
+	return yargs;
+}
+
+/**
+ * Handles the release command.
+ *
+ * @param {object} argv - the arguments passed.
+ */
+export async function releaseCli( argv ) {
+	// Verify we have a valid project.
+	if ( argv.project ) {
+		argv = await parseProj( argv );
+	} else {
+		argv = await promptForProject( argv );
+	}
+
+	// Check if we're working with a beta version.
+	if ( ! argv.beta ) {
+		argv = await promptBeta( argv );
+	}
+
+	// Verify we have a valid script.
+	if ( ! argv.script || argv.script === '' ) {
+		argv = await promptForScript( argv );
+	}
+
+	// Get the info we need for the script.
+	scriptRouter( argv );
```

Comment:
Should we `await` here to ensure the `scriptRouter` has done what it needs to do before proceeding?

Reply:
To my knowledge, the only time we need to use async is when we're [waiting on something](https://developer.mozilla.org/en-US/docs/Learn/JavaScript/Asynchronous/Introducing) to finish that might block the main thread, i.e., writing/reading files, waiting for a prompt, executing a command, etc. Since `scriptRouter` is just assigning variables and returning, it shouldn't be something we need to run asynchronously. However, it also couldn't hurt anything to make it async afaik, in the event we want the router to run something asynchronously in the future. What do you think?

Reply:
Since you declare it async, you should probably await it. If you don't want to await it yet (which would be valid, IMO), then don't declare it async.

Reply:
Ah yeah, I declared it async by mistake thinking something in `scriptRouter` would need something done asynchronously. I'll likely remove the declaration for the function.